### PR TITLE
WIP: Shop redesign

### PR DESCRIPTION
**Pull request by schrockblock**
my Habitica User-ID:

### CODE REVIEW

File: HabitRPG/HRPGTopHeaderNavigationController.m [hunk 1]
```diff
@@ -63,11 +62,11 @@ - (void)viewWillLayoutSubviews {
     [super viewWillLayoutSubviews];
     CGRect parentFrame = self.view.frame;
     self.backgroundView.frame =
-        CGRectMake(0, self.headerYPosition, parentFrame.size.width, self.topHeaderHeight+6);
+        CGRectMake(0, self.headerYPosition, parentFrame.size.width, [self topHeaderHeight]+6);
```
Comment (phillipthelen) [inline HabitRPG/HRPGTopHeaderNavigationController.m]: Why was this changed? I don't think we use this notation to access a property anywhere else.

File: HabitRPG/HRPGTopHeaderNavigationController.m [hunk 2]
```diff
@@ -226,8 +231,17 @@ - (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer
...
 }
 
-- (void)setState:(HRPGTopHeaderState)state {
-    _state = state;
+- (void)setShouldHideTopHeader:(BOOL)shouldHideTopHeader{
+    if (_shouldHideTopHeader != shouldHideTopHeader) {
+        _shouldHideTopHeader = shouldHideTopHeader;
+        if (shouldHideTopHeader) {
+            [self hideHeader];
+        }else{
```
Comment (phillipthelen) [inline HabitRPG/HRPGTopHeaderNavigationController.m]: spaces missing

File: HabitRPG/TableViewController/Inventory/Shops/HRPGShopViewController.h [hunk 3]
```diff
@@ -6,9 +6,9 @@
 //  Copyright © 2016 Phillip Thelen. All rights reserved.
 //
 
-#import "HRPGBaseViewController.h"
+#import "HRPGCollectionViewController.h"
 
-@interface HRPGShopViewController : HRPGBaseViewController <UIAlertViewDelegate>
+@interface HRPGShopViewController : UIViewController <UIAlertViewDelegate>
```
Comment (phillipthelen) [inline HabitRPG/TableViewController/Inventory/Shops/HRPGShopViewController.h]: Is this supposed to extend HRPGCollectionViewController instead? Or why was that imported above?

File: HabitRPG/TableViewController/Modals/HRPGBuyItemModalViewController.swift [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                    HRPGManager.shared().purchaseHourglassItem(shopItem, onSuccess: {
+                        self.dismiss(animated: true, completion: nil)
+                    }, onError: nil)
+                }
+            } else if relevantCurrency == "gems" && !shopItem.canBuy(NSNumber(value: HRPGManager.shared().getUser().balance.floatValue * 4.0)) {
+                let storyboard = UIStoryboard(name: "Main", bundle:nil)
+                let navigationController = storyboard.instantiateViewController(withIdentifier: "PurchaseGemNavController")
+                present(navigationController, animated: true, completion: nil)
+            } else {
+                if relevantCurrency == "gear" {
```
Comment (phillipthelen) [inline HabitRPG/TableViewController/Modals/HRPGBuyItemModalViewController.swift]: Is "gear" correct? That isn't a currency and the method then purchases a quest.

File: HabitRPG/Views/HRPGShopUserHeaderView.swift [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    }()
+    lazy var fetchedResultsController: NSFetchedResultsController<NSFetchRequestResult>? = {
+        return self.createFetchedResultsController()
+    }()
+    lazy var managedObjectContext: NSManagedObjectContext? = self.manager?.getManagedObjectContext()
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(resetUser), name: NSNotification.Name(rawValue: "userChanged"), object: nil)
```
Comment (phillipthelen) [inline HabitRPG/Views/HRPGShopUserHeaderView.swift:30]: This observer also has to be removed before deallocation

File: HabitRPG/Views/HRPGSilverCountView.swift [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+//  HRPGSilverCountView.swift
+//  Habitica
+//
+//  Created by Elliot Schrock on 7/13/17.
+//  Copyright © 2017 Phillip Thelen. All rights reserved.
+//
+
+import UIKit
+
+class HRPGSilverCountView: HRPGCurrencyCountView {
```
Comment (phillipthelen) [inline HabitRPG/Views/HRPGSilverCountView.swift]: Since we're removing silver, this isn't actually needed.

**Review by phillipthelen — CHANGES_REQUESTED** [review]

File: HabitRPG/HRPGTopHeaderNavigationController.m (hunk 1 above)
Comment (schrockblock) [reply HabitRPG/HRPGTopHeaderNavigationController.m]: Good question! I actually removed the property as a whole (see [line 20](https://github.com/schrockblock/habitica-ios/pull/1/files#diff-3cf0a5a938239ec1895e2965bcb54d97L20) above). `topHeaderHeight` is never set, and is instead only ever calculated (see [here](https://github.com/schrockblock/habitica-ios/pull/1/files#diff-3cf0a5a938239ec1895e2965bcb54d97R170)), so I figured it was more of a method than a property. But happy to change it back if you prefer!

File: HabitRPG/HRPGTopHeaderNavigationController.m (hunk 2 above)
Comment (schrockblock) [reply HabitRPG/HRPGTopHeaderNavigationController.m]: Thank you, good catch! I'm sure there are more mistakes like theses, I'll go back and fix them.

File: HabitRPG/TableViewController/Modals/HRPGBuyItemModalViewController.swift (hunk 4 above)
Comment (schrockblock) [reply HabitRPG/TableViewController/Modals/HRPGBuyItemModalViewController.swift]: Absolutely right – typo! Will fix.

File: HabitRPG/TableViewController/Inventory/Shops/HRPGShopViewController.h (hunk 3 above)
Comment (schrockblock) [reply HabitRPG/TableViewController/Inventory/Shops/HRPGShopViewController.h]: I think I originally had this as a `HRPGCollectionViewController` but then decided against it... I'm still working this part out.

File: HabitRPG/Views/HRPGShopUserHeaderView.swift (hunk 5 above)
Comment (schrockblock) [reply HabitRPG/Views/HRPGShopUserHeaderView.swift:30]: Good catch! Thanks!

File: HabitRPG/Views/HRPGSilverCountView.swift (hunk 6 above)
Comment (schrockblock) [reply HabitRPG/Views/HRPGSilverCountView.swift]: Yay! I'll nix it 😄
